### PR TITLE
[MAINT] Format datasets folder

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates #2528

Changes proposed in this pull request:

- [x] update files include in pre-commit config
- [x] update files excluding in pre-commit
- [x] pre-commit run -a
- [x] fix remaining flake8 issues
- [x] remove dead code flagged by vulture
- [ ] apply refactoring from sourcery
- [ ] fix remaining f strings
- [x] update flake8 / isort CI

### CODE REVIEW

File: nilearn/datasets/tests/test_func.py [hunk 1]
```diff
@@ -350,184 +411,156 @@ def test_check_parameters_megatrawls_datasets(request_mocker):
...
-    netmats_data = func.fetch_megatrawls_netmats(data_dir=tmp_path,
-                                                 dimensionality=300,
-                                                 timeseries='multiple_spatial_regression',
-                                                 matrices='full_correlation')
-    assert netmats_data.dimensions == 300
-    assert netmats_data.timeseries == 'multiple_spatial_regression'
-    assert netmats_data.matrices == 'full_correlation'
-
-
-def _cobre_metadata():
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_func.py:413]: removed because unused

File: nilearn/datasets/tests/test_func.py [hunk 2]
```diff
@@ -350,184 +411,156 @@ def test_check_parameters_megatrawls_datasets(request_mocker):
...
-    dummy_data.append({'download_url': 'https://cobre/keys_confounds.json',
-                       'name': 'keys_confounds.json'})
-    dummy_data.append({
-        'download_url': 'https://cobre/keys_phenotypic_data.json',
-        'name': 'keys_phenotypic_data.json'})
-    dummy_data = {'files': dummy_data}
-    return json.dumps(dummy_data), np.asarray(ids_n, dtype='|U17')
-
-
-def _cobre_data(ids):
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_func.py:455]: removed because unused

**Comment by Remi-Gau** [comment]
This is another big one so I am just putting in this PR:
- the automatic reformatting 
- the necessary changes to pacify flake8.

taking of f strings and sourcery refactoring will happen in another PR

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @Remi-Gau !
The diff is pretty hardcore but I didn't see anything problematic (might have been less focused toward the end though... 😅).
If I can find the time, I'll try to take another look (starting from the end this time...).

**Comment by Remi-Gau** [comment]
Given that 95% of the changes come from automated tools that have had the 💩 tested put of them and that the remaining are doc strings fixes to keep flake8 happy I am quite confident that there is no behavior change but I will try to stick to smaller diffs from now on.

File: nilearn/datasets/atlas.py [hunk 3]
```diff
@@ -1273,61 +1396,89 @@ def fetch_atlas_basc_multiscale_2015(data_dir=None, url=None, resume=True,
...
+            'The version of Brain parcellations requested "%s" '
+            "does not exist. Please choose one among them %s."
+            % (version, str(versions))
+        )
 
-    if version == 'sym':
+    if version == "sym":
         url = "https://ndownloader.figshare.com/files/1861819"
-    elif version == 'asym':
+    elif version == "asym":
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/datasets/atlas.py]: done

File: nilearn/datasets/atlas.py [hunk 4]
```diff
@@ -546,104 +598,124 @@ def fetch_atlas_juelich(atlas_name, data_dir=None,
...
 
+def _get_atlas_data_and_labels(
+    atlas_source,
+    atlas_name,
+    symmetric_split=False,
+    data_dir=None,
+    resume=True,
+    verbose=1,
+):
+    """Help for both fetch_atlas_juelich and fetch_atlas_harvard_oxford.
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/datasets/atlas.py]: done

File: nilearn/datasets/tests/test_func.py [hunk 5]
```diff
@@ -111,41 +113,80 @@ def test_fetch_haxby(tmp_path, request_mocker):
...
+        1562298,
+        3205761,
+        3624598,
+    ]
     subs = pd.DataFrame({"Subject": sub1 + sub2 + sub3 + sub4})
     return dict_to_archive(
-        {"ADHD200_40subs_motion_parameters_and_phenotypics.csv":
-         subs.to_csv(index=False)})
+        {
+            "ADHD200_40subs_motion_parameters_and_phenotypics.csv": subs.to_csv(  # noqa E501 line too long and black will not fix it
```
Comment (ymzayek) [inline nilearn/datasets/tests/test_func.py:173]: alternatively you can put the filename in a variable

File: nilearn/datasets/tests/test_func.py [hunk 6]
```diff
@@ -853,37 +897,40 @@ def test_make_spm_auditory_events_file(request_mocker):
...
+    )
 
     assert actual_events_data_string == expected_events_data_string
 
 
 def test_fetch_spm_auditory(request_mocker, tmp_path):
-    import nibabel as nib
     import shutil
+
+    import nibabel as nib
```
Comment (ymzayek) [inline nilearn/datasets/tests/test_func.py:917]: Can these go at the top?

**Review by ymzayek — COMMENTED** [review]
To be continued...

File: nilearn/datasets/tests/test_utils.py [hunk 7]
```diff
@@ -9,44 +8,65 @@
...
-    "dosenbach_2010", "fsaverage3", "fsaverage4", "fsaverage5",
-    "fsaverage6", "fsaverage", "haxby2001",
-    "icbm152_2009", "Megatrawls", "miyawaki2008", "msdl_atlas",
-    "neurovault", "nki_enhanced_surface", "nyu_rest", "oasis1",
-    "pauli_2017", "power_2011", "schaefer_2018", "smith_2009",
-    "talairach_atlas", "yeo_2011"
-])
+datadir = os.path.join(currdir, "data")
+
+DATASET_NAMES = {
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/datasets/tests/test_utils.py:27]: My guess is that this was just missed

File: nilearn/datasets/tests/test_func.py (hunk 5 above)
Comment (Remi-Gau) [reply nilearn/datasets/tests/test_func.py]: good idead

File: nilearn/datasets/tests/test_func.py (hunk 6 above)
Comment (Remi-Gau) [reply nilearn/datasets/tests/test_func.py]: done

File: nilearn/datasets/tests/test_utils.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (bthirion) [reply nilearn/datasets/tests/test_utils.py:27]: Indeed

**Comment by Remi-Gau** [comment]
switching to draft until #3635 is in

**Comment by Remi-Gau** [comment]
if anyone feels that this is too big for a decent review, let me know and I will split this into smaller PRs.

**Comment by Remi-Gau** [comment]
Will turn this into a draft and replace it with a bunch of smaller PRs that are hopefully easier to digest

**Comment by Remi-Gau** [comment]
closing as all the superseding PRs have now been merged

**Pull request closed without merging**